observing/obsstate.py

```diff
@@ -191,7 +216,29 @@ def iterate_max_session_id():
         except TypeError:
             max_session_id = 0
 
-    return str(max_session_id+1)
+    return max_session_id+1
+
+
+def check_and_create_pi(pi_name):
+    """
+    Check if a PI exists in the database, and if not, create one.
+    """
+
+    with connection_factory() as conn:
+        c = conn.cursor()
+        c.execute("SELECT pi_id FROM pis WHERE pi_name = ?", (pi_name,))
+        row = c.fetchone()
+        if row:
+            return row[0]
+        else:
+            c.execute("SELECT MAX(pi_id) FROM pis")
+            try:
+                max_pi_id = int(c.fetchone()[0])
+            except TypeError:
```

Comment:
instead of try catch, maybe...try checking the length of `fetchone()` or a None check or something?